[BUG] Spark 3.4 build fails

**Describe the bug**
A clear and concise description of what the bug is.
```
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:89: overloaded method value checkColumnNameDuplication with alternatives:
  (columnNames: Seq[String],caseSensitiveAnalysis: Boolean)Unit <and>
  (columnNames: Seq[String],resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit
 cannot be applied to (Seq[String], String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:91: overloaded method value checkColumnNameDuplication with alternatives:
  (columnNames: Seq[String],caseSensitiveAnalysis: Boolean)Unit <and>
  (columnNames: Seq[String],resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit
 cannot be applied to (Seq[String], String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:178: overloaded method value checkColumnNameDuplication with alternatives:
  (columnNames: Seq[String],caseSensitiveAnalysis: Boolean)Unit <and>
  (columnNames: Seq[String],resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit
 cannot be applied to (Seq[String], String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:289: overloaded method value checkSchemaColumnNameDuplication with alternatives:
  (schema: org.apache.spark.sql.types.StructType,resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit <and>
  (schema: org.apache.spark.sql.types.DataType,caseSensitiveAnalysis: Boolean)Unit
 cannot be applied to (org.apache.spark.sql.types.StructType, String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:293: overloaded method value checkSchemaColumnNameDuplication with alternatives:
  (schema: org.apache.spark.sql.types.StructType,resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit <and>
  (schema: org.apache.spark.sql.types.DataType,caseSensitiveAnalysis: Boolean)Unit
 cannot be applied to (org.apache.spark.sql.types.StructType, String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala:299: overloaded method value checkSchemaColumnNameDuplication with alternatives:
  (schema: org.apache.spark.sql.types.StructType,resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit <and>
  (schema: org.apache.spark.sql.types.DataType,caseSensitiveAnalysis: Boolean)Unit
 cannot be applied to (org.apache.spark.sql.types.StructType, String, org.apache.spark.sql.catalyst.analysis.Resolver)
[ERROR] [Error] /home/remziy/working/rapids/spark-rapids/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuInsertIntoHadoopFsRelationCommand.scala:58: overloaded method value checkColumnNameDuplication with alternatives:
  (columnNames: Seq[String],caseSensitiveAnalysis: Boolean)Unit <and>
  (columnNames: Seq[String],resolver: org.apache.spark.sql.catalyst.analysis.Resolver)Unit
 cannot be applied to (Seq[String], String, Boolean)
[ERROR] 7 errors found
```
**Steps/Code to reproduce bug**
```
mvn package -Dbuildver=340 -DskipTests -s ./jenkins/settings.xml -f ./pom.xml
```
**Expected behavior**
A clear and concise description of what you expected to happen.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

**Additional context**
Add any other context about the problem here.

--- a/sql-plugin/src/main/311until340-all/scala/org/apache/spark/sql/rapids/shims/SchemaUtilsShims.scala
+++ b/sql-plugin/src/main/311until340-all/scala/org/apache/spark/sql/rapids/shims/SchemaUtilsShims.scala
@@ -1,0 +1,47 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.shims
+
+import org.apache.spark.sql.catalyst.analysis.Resolver
+import org.apache.spark.sql.types.{DataType, StructType}
+import org.apache.spark.sql.util.SchemaUtils
+
+private[spark] object SchemaUtilsShims {
+  def checkSchemaColumnNameDuplication(
+      schema: DataType,
+      colType: String,
+      caseSensitiveAnalysis: Boolean = false): Unit = {
+    SchemaUtils.checkSchemaColumnNameDuplication(schema, colType, caseSensitiveAnalysis)
+  }
+
+  def checkSchemaColumnNameDuplication(
+      schema: StructType,
+      colType: String,
+      resolver: Resolver): Unit = {
+    SchemaUtils.checkSchemaColumnNameDuplication(schema, colType, resolver)
+  }
+
+  def checkColumnNameDuplication(
+      columnNames: Seq[String], colType: String, resolver: Resolver): Unit = {
+    SchemaUtils.checkColumnNameDuplication(columnNames, colType, resolver)
+  }
+
+  def checkColumnNameDuplication(
+      columnNames: Seq[String], colType: String, caseSensitiveAnalysis: Boolean): Unit = {
+    SchemaUtils.checkColumnNameDuplication(columnNames, colType, caseSensitiveAnalysis)
+  }
+}

--- a/sql-plugin/src/main/340+/scala/org/apache/spark/sql/rapids/shims/SchemaUtilsShims.scala
+++ b/sql-plugin/src/main/340+/scala/org/apache/spark/sql/rapids/shims/SchemaUtilsShims.scala
@@ -1,0 +1,47 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.shims
+
+import org.apache.spark.sql.catalyst.analysis.Resolver
+import org.apache.spark.sql.types.{DataType, StructType}
+import org.apache.spark.sql.util.SchemaUtils
+
+private[spark] object SchemaUtilsShims {
+  def checkSchemaColumnNameDuplication(
+      schema: DataType,
+      colType: String,
+      caseSensitiveAnalysis: Boolean = false): Unit = {
+    SchemaUtils.checkSchemaColumnNameDuplication(schema, caseSensitiveAnalysis)
+  }
+
+  def checkSchemaColumnNameDuplication(
+      schema: StructType,
+      colType: String,
+      resolver: Resolver): Unit = {
+    SchemaUtils.checkSchemaColumnNameDuplication(schema, resolver)
+  }
+
+  def checkColumnNameDuplication(
+      columnNames: Seq[String], colType: String, resolver: Resolver): Unit = {
+    SchemaUtils.checkColumnNameDuplication(columnNames, resolver)
+  }
+
+  def checkColumnNameDuplication(
+      columnNames: Seq[String], colType: String, caseSensitiveAnalysis: Boolean): Unit = {
+    SchemaUtils.checkColumnNameDuplication(columnNames, caseSensitiveAnalysis)
+  }
+}

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuDataSource.scala
@@ -47,9 +47,9 @@ import org.apache.spark.sql.execution.datasources.v2.orc.OrcDataSourceV2
 import org.apache.spark.sql.execution.streaming._
 import org.apache.spark.sql.execution.streaming.sources.{RateStreamProvider, TextSocketSourceProvider}
 import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.rapids.shims.SchemaUtilsShims
 import org.apache.spark.sql.sources._
 import org.apache.spark.sql.types.{CalendarIntervalType, DataType, StructType}
-import org.apache.spark.sql.util.SchemaUtils
 import org.apache.spark.util.{HadoopFSUtils, ThreadUtils, Utils}
 
 /**
@@ -86,9 +86,9 @@ case class GpuDataSource(
   }
 
   bucketSpec.foreach { bucket =>
-    SchemaUtils.checkColumnNameDuplication(
+    SchemaUtilsShims.checkColumnNameDuplication(
       bucket.bucketColumnNames, "in the bucket definition", equality)
-    SchemaUtils.checkColumnNameDuplication(
+    SchemaUtilsShims.checkColumnNameDuplication(
       bucket.sortColumnNames, "in the sort definition", equality)
   }
 
@@ -175,7 +175,7 @@ case class GpuDataSource(
     // we have the existing tests for the cases (e.g., `ParquetHadoopFsRelationSuite`).
     // See SPARK-18108 and SPARK-21144 for related discussions.
     try {
-      SchemaUtils.checkColumnNameDuplication(
+      SchemaUtilsShims.checkColumnNameDuplication(
         (dataSchema ++ partitionSchema).map(_.name),
         "in the data schema and the partition schema",
         equality)
@@ -286,17 +286,17 @@ case class GpuDataSource(
 
     relation match {
       case hs: HadoopFsRelation =>
-        SchemaUtils.checkSchemaColumnNameDuplication(
+        SchemaUtilsShims.checkSchemaColumnNameDuplication(
           hs.dataSchema,
           "in the data schema",
           equality)
-        SchemaUtils.checkSchemaColumnNameDuplication(
+        SchemaUtilsShims.checkSchemaColumnNameDuplication(
           hs.partitionSchema,
           "in the partition schema",
            equality)
         DataSourceUtils.verifySchema(hs.fileFormat, hs.dataSchema)
       case _ =>
-        SchemaUtils.checkSchemaColumnNameDuplication(
+        SchemaUtilsShims.checkSchemaColumnNameDuplication(
           relation.schema,
           "in the data schema",
            equality)

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuInsertIntoHadoopFsRelationCommand.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuInsertIntoHadoopFsRelationCommand.scala
@@ -33,7 +33,7 @@ import org.apache.spark.sql.execution.SparkPlan
 import org.apache.spark.sql.execution.command.{AlterTableAddPartitionCommand, AlterTableDropPartitionCommand, CommandUtils}
 import org.apache.spark.sql.execution.datasources.{FileFormatWriter, FileIndex, PartitioningUtils}
 import org.apache.spark.sql.internal.SQLConf.PartitionOverwriteMode
-import org.apache.spark.sql.util.SchemaUtils
+import org.apache.spark.sql.rapids.shims.SchemaUtilsShims
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
 case class GpuInsertIntoHadoopFsRelationCommand(
@@ -55,7 +55,7 @@ case class GpuInsertIntoHadoopFsRelationCommand(
 
   override def runColumnar(sparkSession: SparkSession, child: SparkPlan): Seq[ColumnarBatch] = {
     // Most formats don't do well with duplicate columns, so lets not allow that
-    SchemaUtils.checkColumnNameDuplication(
+    SchemaUtilsShims.checkColumnNameDuplication(
       outputColumnNames,
       s"when inserting into $outputPath",
       sparkSession.sessionState.conf.caseSensitiveAnalysis)

